core/src/layers/prometheus_client.rs

```diff
@@ -82,15 +83,82 @@ use crate::*;
 #[derive(Clone, Debug)]
 pub struct PrometheusClientLayer {
     metrics: PrometheusClientMetricDefinitions,
+    /// The level we will keep in the path label.
+    path_label_level: usize,
 }
 
 impl PrometheusClientLayer {
-    /// Create PrometheusClientLayer while registering itself to this registry. Please keep in caution
-    /// that do NOT call this method multiple times with a same registry. If you want to initialize multiple
-    /// [`PrometheusClientLayer`] with a single registry, you should use [`Arc::clone`] instead.
+    /// Create [`PrometheusClientLayer`] while registering itself to this registry.
+    /// Please keep in caution that do NOT call this method multiple times with a same registry.
+    /// If you want to initialize multiple [`PrometheusClientLayer`] with a single registry,
+    /// you should use [`Arc::clone`] instead.
     pub fn new(registry: &mut Registry) -> Self {
-        let metrics = PrometheusClientMetricDefinitions::register(registry);
-        Self { metrics }
+        Self::builder().build(registry)
+    }
+
+    /// Create a [`PrometheusClientLayerBuilder`].
+    pub fn builder() -> PrometheusClientLayerBuilder {
```

Comment:
Hi, we usually avoid providing too many structures at the layer level. How about just exposing the API at the `PrometheusClientLayer` level?

Reply:
The design of the `prometheus` library and the `prometheus-client` library is quite different.

For `prometheus-client` library, we have to accept the `&mut Registry` type to register our configured metrics.
If we want to provide some configurable methods similar to the `PrometheusLayer`, it seems that we have to construct the configuration through an additional type. 

Do you have any other good ideas?

BTW, for `prometheus` library, we could expose the configurable API at the `PrometheusLayer` level.